ubuntu-checkpatch: add upstream proveance check

corytodd opened the pull request:
If the declared provenance is one of the common upstreams, check that the referenced commit exists using an HTTP request.

I know you said to stick to the stdlib but requests seems like a fair requirement here.

Commit:
ubuntu-checkpatch: add upstream proveance check

If the declared provenance is one of the common upstreams, check that
the referenced commit exists using an HTTP request.

Signed-off-by: Cory Todd <cory.todd@canonical.com>

juergh commented on ubuntu-checkpatch:
```diff
@@ -362,8 +365,24 @@ class CheckCherrypickBackport(Check):
+        if commit and self.patch.provenance in ("", "linux-next", "linux-stable"):
+            linux_base = "https://git.kernel.org/pub/scm/linux/kernel/git"
+            upstream = {
+                "": f"{linux_base}/torvalds/linux.git/commit/?id={commit}",
```
Can we stick to format() instead of f-string? This script might be run on older releases with an older Python.

corytodd replied:
yes, done.

juergh commented on ubuntu-checkpatch:
```diff
@@ -362,8 +365,24 @@ class CheckCherrypickBackport(Check):
+                self.result(FAIL, "declared provenance does not contain referenced commit")
+                rc = 1
+            else:
+                self.result(WARN, f"unable to check upstream for commit: {upstream} returned HTTP_STATUS={r.status_code}")
```
Warnings are supposed to be non-fatal so this should be FAIL.

Maybe we should introduce an offline mode that skips all checks that require an internet connection?

corytodd replied:
Marked as FAIL.

Offline mode would be good. It might make sense to have a SKIP state too then. I'll work up another PR for that.

juergh reviewed (changes requested):
I like it, thanks! A couple of comments inline...

Commit:
ubuntu-checkpatch: use .format instead of f-strings

Users may be using Python <=3.5 so don't use f-strings.

Signed-off-by: Cory Todd <cory.todd@canonical.com>

Commit:
ubuntu-checkpatch: use FAIL for fatal errors

WARN is reserved for non-fatal problems. Not being able to check for
commit provenance is in fact fatal.

Signed-off-by: Cory Todd <cory.todd@canonical.com>
